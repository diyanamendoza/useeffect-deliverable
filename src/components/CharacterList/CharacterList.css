ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    /* justify-content: space-evenly; */
    width: 800px;
}

li {
    list-style-type: none;
    width: 400px;
}

img {
    border-radius: 30px;
}