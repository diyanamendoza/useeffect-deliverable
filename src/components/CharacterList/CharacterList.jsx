import Character from "../Character/Character"
import './CharacterList.css'


export default function CharacterList({ characters }) {
    return (
      <div className="character-list-container">
        {(characters.length === 0) ? <h2>Poop. No results to show!</h2> :
      <ul aria-label="characterList" className="characterList">
        {characters.map((character) => {
          return (
            <li className="listItem" key={character.name + Math.random() * 100}>
              <Character character={character} />
            </li>
          )
        })}
      </ul>}
      </div>
    )
  }