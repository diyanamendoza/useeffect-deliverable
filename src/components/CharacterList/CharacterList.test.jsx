import { render } from '@testing-library/react';
import CharacterList from './CharacterList.jsx';

it('should render character list component', () => {
  const { container } = render(
    <CharacterList characters={[{name: 'Ted', origin: 'Wisconsin', species: 'Rodent', gender: 'Unknown', image: 'https://hips.hearstapps.com/digitalspyuk.cdnds.net/17/20/1495207668-keanu-reeves-bill-bill-and-ted.jpg', status: 'Coasting'}, {name: 'Bill', origin: 'Wisconsin', species: 'Rodent', gender: 'Unknown', image: 'https://upload.wikimedia.org/wikipedia/commons/thumb/f/f1/Bill_Murray-8882.jpg/440px-Bill_Murray-8882.jpg', status: 'Coasting'}]} />
  );
  expect(container).toMatchSnapshot();
});