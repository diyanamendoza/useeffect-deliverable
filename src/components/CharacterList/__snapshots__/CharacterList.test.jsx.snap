// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render character list component 1`] = `
<div>
  <div
    class="character-list-container"
  >
    <ul
      aria-label="characterList"
      class="characterList"
    >
      <li
        class="listItem"
      >
        <figure
          aria-label="character"
        >
          <img
            alt="Ted-Rodent"
            src="https://hips.hearstapps.com/digitalspyuk.cdnds.net/17/20/1495207668-keanu-reeves-bill-bill-and-ted.jpg"
          />
          <div
            class="details"
          >
            <h3>
              Ted
            </h3>
            <p>
              Rodent
               | 
              Unknown
            </p>
            <p>
              Origin: 
              Wisconsin
            </p>
            <p>
              Status: 
              Coasting
            </p>
          </div>
        </figure>
      </li>
      <li
        class="listItem"
      >
        <figure
          aria-label="character"
        >
          <img
            alt="Bill-Rodent"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/f1/Bill_Murray-8882.jpg/440px-Bill_Murray-8882.jpg"
          />
          <div
            class="details"
          >
            <h3>
              Bill
            </h3>
            <p>
              Rodent
               | 
              Unknown
            </p>
            <p>
              Origin: 
              Wisconsin
            </p>
            <p>
              Status: 
              Coasting
            </p>
          </div>
        </figure>
      </li>
    </ul>
  </div>
</div>
`;
